lab-adan/test/server-test.js

```diff
@@ -0,0 +1,104 @@
+'use strict';
+
+const superagent = require('superagent');
+const expect = require('expect');
+const server = require('../lib/server.js');
+const GameScore = require('../model/gamescore.js');
+const fs = require('fs-extra');
+const PORT = process.env.PORT || 3000;
+
+describe('/api/gamescore routes', () => {
+  let testGame;
+
+  before(done => {
+    fs.ensureDir(`${__dirname}/../data`)
+    .then(() => {
+      server.listen(PORT, () => done());
+    });
+  });
+
+  after(done => {
+    fs.emptyDir(`${__dirname}/../data`)
+    .then(() => server.close(() => done()));
+  });
+
+  describe('GameScore Constructor', () => {
+    it('Should return a GameScore Object', done => {
+      new GameScore('tetris', '9000', 1)
+        .save()
+        .then(data =>{
+          expect(data.gameName).toEqual('tetris');
+          expect(data.score).toEqual('9000');
+          expect(data.id).toEqual(1);
```

Comment:
This is failing because you are supplying the ID to the constructor but then changing the value assigned to that id property in your storage module before setting it to storage.